File: tests/testdata/codegen/azure-native-v2-eventgrid-pp/dotnet/azure-native-v2-eventgrid.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using AzureNative = Pulumi.AzureNative;

return await Deployment.RunAsync(() => 
{
    var example = new AzureNative.EventGrid.EventSubscription("example", new()
    {
        Destination = new AzureNative.EventGrid.Inputs.EventHubEventSubscriptionDestinationArgs
        {
            EndpointType = "EventHub",
            ResourceId = "example",
        },
        ExpirationTimeUtc = "example",
        Scope = "example",
    });

});

File: tests/testdata/codegen/azure-native-v2-eventgrid-pp/go/azure-native-v2-eventgrid.go
package main

import (
	eventgrid "github.com/pulumi/pulumi-azure-native-sdk/eventgrid/v2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := eventgrid.NewEventSubscription(ctx, "example", &eventgrid.EventSubscriptionArgs{
			Destination: &eventgrid.EventHubEventSubscriptionDestinationArgs{
				EndpointType: pulumi.String("EventHub"),
				ResourceId:   "example",
			},
			ExpirationTimeUtc: "example",
			Scope:             pulumi.String("example"),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
